File: LeetCode/2962.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        max_elem = max(nums)
        cnt = 0
        left, ans = 0, 0
        for right, elem in enumerate(nums):
            if nums[right] == max_elem:
                cnt += 1
            while cnt >= k:
                if nums[left] == max_elem:
                    cnt -= 1
                left += 1
            ans += left
        return ans

File: LeetCode/2962.cpp
class Solution {
public:
	long long countSubarrays(vector<int> &nums, int k) {
		long long ans = 0;
		int target = *std::ranges::max_element(nums), n = nums.size(), cnt = 0;
		for (int left = 0, right = 0; right < n; right++) {
			cnt += (nums[right] == target);
			while (cnt >= k) {
				cnt -= (nums[left] == target);
				++left;
			}
			ans += left;
		}
		return ans;
	}
};
